pgjdbc/src/main/java/org/postgresql/ds/common/BaseDataSource.java

```diff
@@ -758,6 +759,22 @@ public void setSslRootCert(@Nullable String file) {
     PGProperty.SSL_ROOT_CERT.set(properties, file);
   }
 
+  /**
+   * @param sslNegotiation one of SSLNegotiation.POSTGRES or SSLNegotiation.DIRECT
+   * @see PGProperty#SSL_NEGOTIATION
+   */
+  public void setSslNegotiation(@Nullable SSLNegotiation sslNegotiation) {
+    PGProperty.SSL_NEGOTIATION.set(properties, sslNegotiation.value());
+  }
+
+  /**
+   * @return SSL Negotiation scheme
+   * @see PGProperty#SSL_NEGOTIATION
+   */
+  public @Nullable SSLNegotiation getSslNegotiation() {
```

Comment:
The default value is not nullable, so let's use non-nullable for the getter:
```suggestion
  public SSLNegotiation getSslNegotiation() {
```